src/Acuminator/Acuminator.Analyzers/Analyzers/DAC/DacPropertyAttributes/DacExtensionDefaultAttributeAnalyzer.cs

```diff
@@ -0,0 +1,153 @@
+				{
+					var typesHierarchy = attribute.AttributeClass.GetBaseTypesAndThis();
+					if(typesHierarchy.Contains(pxContext.AttributeTypes.PXDefaultAttribute))
+					{
+						foreach (var argument in attribute.NamedArguments)
+						{
+							if (argument.Key.Contains("PersistingCheck") && (int)argument.Value.Value == (int)PXPersistingCheck.Nothing)
+								return;
+						}
+						Location[] locations = await Task.WhenAll(GetAttributeLocationAsync(attribute, symbolContext.CancellationToken));
+						Location attributeLocation = locations[0];
+
+						if (attributeLocation != null)
+						{
+							var diagnosticProperties = new Dictionary<string, string>{
+															{ DiagnosticProperty.IsBoundField,
+															  isBoundField.ToString() }}.ToImmutableDictionary();
+
+							symbolContext.ReportDiagnostic(
+								Diagnostic.Create(
+									Descriptors.PX1030_DefaultAttibuteToExisitingRecords, attributeLocation, diagnosticProperties));
+						}
+					}
+				}
+			}
+			else
+			{
+				foreach (var attribute in attributes)
+				{
+					var typesHierarchy = attribute.AttributeClass.GetBaseTypesAndThis();
+					if(typesHierarchy.Contains(pxContext.AttributeTypes.PXDefaultAttribute) &&
+						!typesHierarchy.Contains(pxContext.AttributeTypes.PXUnboundDefaultAttribute))
+					{
+						Location[] locations = await Task.WhenAll(GetAttributeLocationAsync(attribute, symbolContext.CancellationToken));
+						Location attributeLocation = locations[0];
+
+						if (attributeLocation != null)
+						{
+							var diagnosticProperties = new Dictionary<string, string>
+						{
+							{ DiagnosticProperty.IsBoundField, isBoundField.ToString() }
+						}.ToImmutableDictionary();
+
+							symbolContext.ReportDiagnostic(
+								Diagnostic.Create(
+									Descriptors.PX1030_DefaultAttibuteToExisitingRecords, attributeLocation, diagnosticProperties));
+						}
+					}
+
+				}
+			}
+			return;
+		}
+
+		private static bool CheckProperty(IPropertySymbol property, PXContext pxContext)
+		{
+			var parent = property?.ContainingType;
+
+			if (parent == null || (!parent.ImplementsInterface(pxContext.IBqlTableType) && !parent.InheritsFrom(pxContext.PXCacheExtensionType)))
+				return false;
+			return property.Type.TypeKind == TypeKind.Struct ||
+				   property.Type.TypeKind == TypeKind.Class ||
+				   property.Type.TypeKind == TypeKind.Array;
+		}
+
+		private static bool IsIBqlTableTypeImplementation(IPropertySymbol property, PXContext pxContext)
+		{
+			var parent = property?.ContainingType;
+
+			if (parent == null || !parent.ImplementsInterface(pxContext.IBqlTableType))
+				return false;
+			return true;
+		}
+
+
+		public static async Task<Location> GetAttributeLocationAsync(AttributeData attribute, CancellationToken cancellationToken)
+		{
+			SyntaxNode attributeSyntaxNode = null;
+
+			try
+			{
+				attributeSyntaxNode = await attribute.ApplicationSyntaxReference.GetSyntaxAsync(cancellationToken)
```

Comment:
I suggest using `cancellationToken.ThrowIfCancellationRequested` inside `GetSyntaxAsync` and avoid try/catch here.

Reply:
@dropsonic, this was probably copied from field attributes analyzer, the reason behind try/catch is to intercept possible errors because I don't know if GetSyntaxAsync could throw exception. The OperationCancelled exception was added here to distinguidh between cancel and error - in case of error we probably should log it in the future